File: src/OB.Data/Entities/UserAccount.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace OB.Data.Entities
{
    public class UserAccount
    {
        // ----------------------------------------------  
        [Key]
        public int Id { get; set; }

        [StringLength(32)]
        public string Username { get; set; }

        [StringLength(128)]
        public string Password { get; set; }

        [StringLength(128)]
        public string EmailAddress { get; set; }

        public int LoginCount { get; set; }

        [Required]
        public DateTime LastLogin { get; set; }

        [Required]
        public DateTime CreatedOn { get; set; }

        [Required, StringLength(32)]
        public DateTime CreatedBy { get; set; }

        public DateTime? UpdatedOn { get; set; }

        [StringLength(32)]
        public DateTime UpdatedBy { get; set; }
        // ----------------------------------------------  
        public virtual List<UserAccountRole> UserAccountRoles { get; set; }

    }
}

File: _assets/output_gen/jsModels/UserAccount.js

//---------------------------------------------------------------------------

export default class UserAccount {
  constructor(options) {

    this.Id = null;
    this.Username = null;
    this.Password = null;
    this.EmailAddress = null;
    this.LoginCount = null;
    this.LastLogin = null;
    this.DateCreated = null;
    this.DateUpdated = null;
    this.UpdatedBy = null;
    this.UserAccountRoles = null;

    if(options) {
      this.init(options);
    }
  }

  //---------------------------------------------------------------------------
  init(options) {

    this.Id = options.Id || this.Id;
    this.Username = options.Username || this.Username;
    this.Password = options.Password || this.Password;
    this.EmailAddress = options.EmailAddress || this.EmailAddress;
    this.LoginCount = options.LoginCount || this.LoginCount;
    this.LastLogin = options.LastLogin || this.LastLogin;
    this.DateCreated = options.DateCreated || this.DateCreated;
    this.DateUpdated = options.DateUpdated || this.DateUpdated;
    this.UpdatedBy = options.UpdatedBy || this.UpdatedBy;
    this.UserAccountRoles = options.UserAccountRoles || this.UserAccountRoles;

  }  
  //---------------------------------------------------------------------------  
}